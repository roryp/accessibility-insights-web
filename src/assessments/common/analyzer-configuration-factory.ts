// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Messages } from 'common/messages';
import { RuleAnalyzerConfiguration } from 'injected/analyzers/analyzer';
import { ScannerUtils } from 'injected/scanner-utils';

export class AnalyzerConfigurationFactory {
    public static forScanner(base: Partial<RuleAnalyzerConfiguration>): RuleAnalyzerConfiguration {
        const defaultValues: Partial<RuleAnalyzerConfiguration> = {
            resultProcessor: (scanner: ScannerUtils) => scanner.getAllCompletedInstances,
            telemetryProcessor: telemetryFactory => telemetryFactory.forAssessmentRequirementScan,
            analyzerMessageType: Messages.Assessment.AssessmentScanCompleted,
        };

        return {
            ...defaultValues,
            ...base,
        } as RuleAnalyzerConfiguration;
    }
}
