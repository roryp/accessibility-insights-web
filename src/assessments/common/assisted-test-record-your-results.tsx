// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export const AssistedTestRecordYourResults = () => (
    <li>
        Record your results:
        <ol>
            <li>
                Select <Markup.Term>Fail</Markup.Term> for any instances that do not meet the
                requirement.
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>. Or, after you have marked all
                failures, select <Markup.Term>Pass unmarked instances.</Markup.Term>
            </li>
        </ol>
    </li>
);
