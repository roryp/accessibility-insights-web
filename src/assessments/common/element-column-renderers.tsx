// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { AssessmentInstanceDetailsColumn } from 'DetailsView/components/assessment-instance-details-column';
import * as React from 'react';

export function onRenderPathColumn(item: InstanceTableRow): JSX.Element {
    let textContent = '';
    if (item.instance.target) {
        textContent = item.instance.target.join(';');
    }

    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={textContent}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}

export function onRenderSnippetColumn(item: InstanceTableRow): JSX.Element {
    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={item.instance.html}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}
