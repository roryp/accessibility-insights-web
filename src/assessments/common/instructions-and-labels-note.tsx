// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import * as Markup from '../markup';

export const InstructionsAndLabelsNotes = () => (
    <Markup.Emphasis>
        Note: For WCAG 2.1 compliance, both visible instructions and labels must be programmatically
        related to a widget. For WCAG 2.0 compliance, only visible instructions must be
        programmatically related to a widget.
    </Markup.Emphasis>
);
