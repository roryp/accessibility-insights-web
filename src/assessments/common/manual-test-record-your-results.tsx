// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export interface ManualTestRecordYourResultsProps {
    isMultipleFailurePossible: boolean;
}

export const ManualTestRecordYourResults = (props: ManualTestRecordYourResultsProps) => (
    <li>
        Record your results:
        <ol>
            <li>
                If you find {props.isMultipleFailurePossible ? 'any failures' : 'a failure'}, select{' '}
                <Markup.Term>Fail</Markup.Term>,
                {props.isMultipleFailurePossible
                    ? ' then add them as failure instances'
                    : ' then add the failure instance'}
                .
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>.
            </li>
        </ol>
    </li>
);
