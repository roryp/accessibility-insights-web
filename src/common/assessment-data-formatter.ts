// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';

export class AssessmentDataFormatter {
    public formatAssessmentData = (assessmentData: AssessmentStoreData): string => {
        const versionedData = this.addVersionNumber(assessmentData);
        const assessmentReportData = JSON.stringify(versionedData);
        return assessmentReportData;
    };

    private addVersionNumber = (assessmentData: AssessmentStoreData): VersionedAssessmentData => {
        const versionedAssessmentData = {
            version: 1,
            assessmentData: assessmentData,
        };
        return versionedAssessmentData;
    };
}
