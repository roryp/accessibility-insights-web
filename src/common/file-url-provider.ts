// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { WindowUtils } from './window-utils';

export class FileURLProvider {
    constructor(
        private windowUtils: WindowUtils,
        private provideBlob: (blobParts: any[], mimeType: string) => Blob,
    ) {}

    public provideURL = (blobParts: any[], mimeType: string): string => {
        const blob = this.provideBlob(blobParts, mimeType);
        const blobURL = this.windowUtils.createObjectURL(blob);
        return blobURL;
    };
}
