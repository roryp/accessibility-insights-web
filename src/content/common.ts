// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { ContentCreator } from 'views/content/content-page';
import { GuidanceTitle as gt } from 'views/content/guidance-title';
import { link } from './link';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
