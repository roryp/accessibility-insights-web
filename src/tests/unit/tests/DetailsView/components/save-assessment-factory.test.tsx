// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentDataFormatter } from 'common/assessment-data-formatter';
import { FileURLProvider } from 'common/file-url-provider';
import { FileNameBuilder } from 'common/filename-builder';
import {
    AssessmentStoreData,
    AssessmentData,
} from 'common/types/store-data/assessment-result-data';
import {
    getSaveButtonForAssessment,
    getSaveButtonForFastPass,
    SaveAssessmentFactoryDeps,
    SaveAssessmentFactoryProps,
} from 'DetailsView/components/save-assessment-factory';
import { IMock, Mock, Times } from 'typemoq';

describe('SaveAssessmentFactory', () => {
    let deps: SaveAssessmentFactoryDeps;
    let props: SaveAssessmentFactoryProps;
    const fileURLProviderMock = Mock.ofType(FileURLProvider);
    const assessmentDataFormatterMock = Mock.ofType(AssessmentDataFormatter);
    const fileNameBuilderMock = Mock.ofType(FileNameBuilder);
    const getCurrentDateStub = () => dateValue;
    const dateValue = new Date(2020, 11, 20);

    beforeEach(() => {
        const assessmentStoreData = {
            assessments: {
                ['assessment-1']: {
                    fullAxeResultsMap: null,
                    generatedAssessmentInstancesMap: null,
                    manualTestStepResultMap: {
                        ['assessment-1-step-1']: {
                            instances: [],
                            status: 2,
                            id: 'assessment-1-step-1',
                        },
                        ['removed-step']: {
                            instances: [],
                            status: 2,
                            id: '123',
                        },
                    },
                    testStepStatus: {},
                },
            } as { [key: string]: AssessmentData },
            persistedTabInfo: { title: 'SavedAssessment123' },
        } as AssessmentStoreData;

        deps = {
            assessmentDataFormatter: assessmentDataFormatterMock.object,
            fileURLProvider: fileURLProviderMock.object,
            fileNameBuilder: fileNameBuilderMock.object,
            getCurrentDate: getCurrentDateStub,
        } as SaveAssessmentFactoryDeps;
        props = {
            deps,
            assessmentStoreData,
        } as SaveAssessmentFactoryProps;
    });

    describe('getSaveButtonForAssessment', () => {
        test('renders save assessment button', () => {
            const assessmentData = props.assessmentStoreData;
            const formattedAssessmentData = JSON.stringify(assessmentData);
            const title = props.assessmentStoreData.persistedTabInfo.title;

            assessmentDataFormatterMock
                .setup(a => a.formatAssessmentData(assessmentData))
                .returns(() => formattedAssessmentData)
                .verifiable(Times.once());

            fileURLProviderMock
                .setup(f => f.provideURL([formattedAssessmentData], 'application/json'))
                .returns(() => 'fileURL')
                .verifiable(Times.once());

            fileNameBuilderMock.setup(f => f.getDateSegment(dateValue)).returns(() => 'date');

            fileNameBuilderMock.setup(f => f.getTitleSegment(title)).returns(() => 'title');

            const rendered = getSaveButtonForAssessment(props);
            expect(rendered).toMatchSnapshot();
            assessmentDataFormatterMock.verifyAll();
            fileURLProviderMock.verifyAll();
            fileNameBuilderMock.verifyAll();
        });
    });

    describe('getSaveButtonForFastPass', () => {
        test('renders save assessment button as null', () => {
            const rendered = getSaveButtonForFastPass(props);
            expect(rendered).toBeNull();
        });
    });
});
