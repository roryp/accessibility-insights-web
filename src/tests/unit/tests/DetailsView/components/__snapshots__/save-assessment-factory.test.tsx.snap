// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveAssessmentFactory getSaveButtonForAssessment renders save assessment button 1`] = `
<SaveAssessmentButton
  download="SavedAssessment_date_title.a11ywebassessment"
  href="fileURL"
/>
`;
